flit/__init__.py

```diff
@@ -16,25 +16,34 @@
 log = logging.getLogger(__name__)
 
 
-class PythonNotFoundError(Exception): pass
+class PythonNotFoundError(FileNotFoundError): pass
 
-def find_python_executable(python):
+
+def find_python_executable(python: Optional[str] = None) -> str:
```

Comment:
I thought I remembered that `=None` implicitly made a parameter optional, which sent me down a rabbit hole to find where I had read it. I had remembered right, but the shorthand is no longer recommended, and the rabbit hole ended at https://github.com/python/peps/pull/1345 .

I'm slightly sad about that, because I think things like this would be clearer with it. But that's how it went, so this can stay as is.

Reply:
Interesting bit of history on `typing`. It is rather annoying that, would be some nice sugar. Oh well!